### ISSUE TITLE
Monotonicity of expectiles

### ISSUE THREAD

**Issue by joshsilverbeck1** [StatMixedML/LightGBMLSS#7]
Hi,

Thanks for the great package.

I am trying to predict the expectiles of a distribution. I understood that because this approach models the entire distribution, and because the true expectiles of any distribution should be monotonic, the predictions would also be monotonic. However, I'm not getting this property. Why might this be / is there anyway to force monotonicity?

Thanks

Josh

**Comment by StatMixedML** [comment]
Hi @joshsilverbeck1,

Thanks for your interest in project.

You are right. As with quantiles, expectiles are derived from a monotonically increasing cumulative distribution function. As such, theoretical expectiles are increasing / non-crossing. However, as with quantiles in the base lightgbm/xgboost implementations, expectiles can cross in practical applications, especially if a dense set of expectiles is modeled. This is because each expectile is modeled individually. In its current implementations, unfortunately, there is no constraint included that prevents expectiles from crossing. 

There is some literature on this, yet, I cannot recall the exact papers (you can search for, e.g., quantile sheets). One could also add a penalty term to the loss function, which, during training, ensures monotonicity. However, for predictions, non-crossing expectiles is still not guaranteed. Also, you would need to modify gradients/hessians for this. Easiest way to do this would be to switch from analytical derivatives to PyTorch-autograd for all distributions, which I am currently working on. Yet, this might take some time  ...

Would that answer your question?

**Comment by joshsilverbeck1** [comment]
Thanks for your prompt answer, this is helpful to know :)
